tests/unit-tests/test_nn_encoder.py

```diff
@@ -0,0 +1,296 @@
+        for input_nodes, _, blocks in dataloader:
+            nfeats = gdata.get_node_feats(input_nodes, nfeat_fields, device=dev)
+            efeats_list = gdata.get_blocks_edge_feats(blocks, efeat_fields, device=dev)
+
+        nfeats = generate_dummy_features(nfeats, input_dim, feat_pattern='random', device=dev)
+        efeats_list = [generate_dummy_features(efeats, input_dim, feat_pattern='random', device=dev) \
+                       for efeats in efeats_list]
+
+        blocks = [block.to(dev) for block in blocks]
+
+        encoder = RelationalGCNEncoder(gdata.g,
+                                       input_dim, output_dim,
+                                       num_hidden_layers=len(fanout)-1,
+                                       edge_feat_name=efeat_fields,
+                                       edge_feat_mp_op='concat')
+        assert len(encoder.layers) == 1
+        assert isinstance(encoder.layers[0].conv._get_module(('n0', 'r0', 'n1')),
+                          GraphConvwithEdgeFeat)
+        assert isinstance(encoder.layers[0].conv._get_module(('n0', 'r1', 'n1')),
+                          GraphConvwithEdgeFeat)
+        assert isinstance(encoder.layers[0].conv._get_module(('n1', 'r2', 'n0')),
+                          dgl.nn.GraphConv)
+        encoder = encoder.to(dev)
+        emb5 = encoder(blocks, nfeats, efeats_list)
+        assert emb5['n0'].shape[-1] == output_dim
+        assert emb5['n1'].shape[-1] == output_dim
+
+        # Test 6: normal case, same as case 1, but 3 layer of GNN
+        nfeat_fields = {'n0':['feat'], 'n1': ['feat']}
+        efeat_fields = {('n0', 'r0', 'n1'): ['feat'], ('n0', 'r1', 'n1'): ['feat']}
+
+        fanout = [100, 100, 100]
+        target_idx = {('n0', 'r1', 'n1'): [0, 1]}
+        dataloader = GSgnnEdgeDataLoader(gdata, target_idx, fanout, 10,
+                                        label_field='label',
+                                        train_task=False, remove_target_edge_type=False)
+        for input_nodes, _, blocks in dataloader:
+            nfeats = gdata.get_node_feats(input_nodes, nfeat_fields, device=dev)
+            efeats_list = gdata.get_blocks_edge_feats(blocks, efeat_fields, device=dev)
+
+        nfeats = generate_dummy_features(nfeats, input_dim, feat_pattern='random', device=dev)
+        efeats_list = [generate_dummy_features(efeats, input_dim, feat_pattern='random', device=dev) \
+                       for efeats in efeats_list]
+
+        blocks = [block.to(dev) for block in blocks]
+
+        encoder = RelationalGCNEncoder(gdata.g,
+                                       input_dim, output_dim,
+                                       num_hidden_layers=len(fanout)-1,
+                                       edge_feat_name=efeat_fields,
+                                       edge_feat_mp_op='concat')
+        assert len(encoder.layers) == 3
+        assert isinstance(encoder.layers[0].conv._get_module(('n0', 'r0', 'n1')),
```

Comment:
Shall we also check layers[1] and layers[2]?

Reply:
added.